serializr.d.ts

```diff
@@ -10,8 +10,8 @@ export interface Context {
 export type Factory<T> = (context: Context) => T
 
 export interface PropSchema {
-    serializer(sourcePropertyValue: any);
-    deserializer(jsonValue: any, callback: (err, targetPropertyValue: any) => void, context: Context, currentPropertyValue: any);
+    serializer(sourcePropertyValue: any): any;
+    deserializer(jsonValue: any, callback: (err: any, targetPropertyValue: any) => void, context: Context, currentPropertyValue: any): any;
```

Comment:
yes, deserializer supports async stuff (so additional data can be loaded for example), hence it get's a callback instead of returning something. Serializer is ATM always synchronous (far less usecases to have serialization synchronous, but there are some see #5 ), hence the difference in return type